CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+# Set minimum CMake version to latest version on Gadi:
+cmake_minimum_required(VERSION 3.24.2)
+
+project(
+    CABLE
+    LANGUAGES Fortran
+)
+
+option(CABLE_MPI "Build the MPI executable" OFF)
+
+if(CABLE_MPI)
+    # The MPI implementation is currently not supported in CABLE-POP_TRENDY:
+    message(FATAL_ERROR "MPI compilation not supported.")
+endif()
+
+# third party libs
+if(CABLE_MPI)
+    find_package(MPI REQUIRED COMPONENTS Fortran)
+endif()
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(NETCDF REQUIRED IMPORTED_TARGET "netcdf-fortran")
+
+set(CABLE_Intel_Fortran_FLAGS -fp-model precise -fpp -nofixed -assume byterecl -D__INTEL__ -D__INTEL_COMPILER__ -D__NETCDF3__)
+set(CABLE_Intel_Fortran_FLAGS_DEBUG -O0 -g -traceback -fpe0 -debug extended -check all -warn all -fp-stack-check)
+set(CABLE_Intel_Fortran_FLAGS_RELEASE -O3 -ip)
```

Comment:
These are my compiler flags. There is nothing fancy. I compared them to the main branch
```
set(CABLE_Intel_Fortran_FLAGS -fp-model precise)
set(CABLE_Intel_Fortran_FLAGS_DEBUG -O0 -g -traceback -fpe0)
set(CABLE_Intel_Fortran_FLAGS_RELEASE -O2)

```
 and I was wondering why the main branch does not have these switches. The debug version did not exist before so why not putting all checks and warnings? I also do not know why main has only -O2.

Reply:
The flags in the main branch were left as is when doing the transition from Makefile to CMake (with the exception of https://github.com/CABLE-LSM/CABLE/issues/198 and https://github.com/CABLE-LSM/CABLE/issues/196). It is difficult to track down the origin of these flags in the commit history so I do not know why the flags are the way they are.

@mcuntz we should continue this discussion in this issue thread: https://github.com/CABLE-LSM/CABLE/issues/268 as this focuses on the merging of build configurations between main and CABLE-POP_TRENDY.

I would imagine including most of these flags in the main branch would be trivial but some flags will require testing.

Reply:
I would rather not comment in #268 but will do in #307 .